confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/macros/AbstractMacroConverter.java

```diff
@@ -72,12 +116,48 @@ protected Map<String, String> toXWikiParameters(String confluenceId, Map<String,
     protected String toXWikiParameterName(String confluenceParameterName, String id,
         Map<String, String> confluenceParameters, String confluenceContent)
     {
-        return confluenceParameterName;
+        String name;
+        if (confluenceParameterName.startsWith(USER_PARAMETER_PREFIX)) {
```

Comment:
I'm not a huge fan of the concept of prefix based typing of parameters, I guess it's also a breaking change for any existing macro converter (even if there is probably not that many outside of this module and the confluence migrator pro you are working on). But yes there are not really any other way to type them.

What might be cleaner (also for other use cases) as an alternative would be to move the reference conversion for these types of references to the syntax parser but optionally. What I can think of is some new `ConfluenceReferenceConverter` property (I guess a single one with a method for each type of reference like user, document, etc. should do) in `ConfluenceXHTMLInputProperties` in charge of that (null by default) and implemented and set on confluence-xml side.

Reply:
> Note that @trrenty had started this "prefix work" at https://github.com/xwiki-contrib/confluence/commit/9e19e7c99e386f3956f9e51c95e1191bee47a408

Indeed, looks like I missed it (or probably more accurately was too lazy to focus on the details back then).